<?xml version="1.0" ?>
<sdf version="1.7">
  <world name="sortabot_world">

    <!-- ROS 2 integration plugin -->
    <plugin
      filename="libgz-sim-ros2-system.so"
      name="gz::sim::systems::Ros2System">
      <ros>
        <!-- Use global namespace -->
        <namespace>/</namespace>
        <use_sim_time>true</use_sim_time>
      </ros>
    </plugin>

    <!-- Basic light -->
    <light type="directional" name="sun">
      <pose>0 0 10 0 0 0</pose>
      <diffuse>1.0 1.0 0.9 1</diffuse>
      <specular>0.5 0.5 0.5 1</specular>
      <direction>0.1 0.1 -1</direction>
    </light>

    <!-- Simple ground plane -->
    <model name="ground">
      <static>true</static>
      <link name="link">
        <pose>0 0 -0.05 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <box>
              <size>20 20 0.1</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <box>
              <size>20 20 0.1</size>
            </box>
          </geometry>
          <material>
            <ambient>0.4 0.4 0.4 1</ambient>
            <diffuse>0.7 0.7 0.7 1</diffuse>
          </material>
        </visual>
      </link>
    </model>

    <!-- Your robot -->
    <include>
      <uri>file:///home/ellynrose/Robotics_Assignment/src/sortabot/models/sortabot.sdf</uri>
      <name>sortabot</name>
      <pose>0 0 0.25 0 0 0</pose>
    </include>

  </world>
</sdf>
